From f84d99e2b2d34004176afc42ac44faa1569f83e7 Mon Sep 17 00:00:00 2001
From: Andrew Liu <shengping.liu@windriver.com>
Date: Thu, 9 Sep 2010 16:42:46 +0800
Subject: [PATCH 180/252] crypto: caam - algapi: fix DEBUG build

Patch taken from FSL vendor SDK 2.1.

this fixes building algapi when DEBUG is turned on.

Signed-off-by: Kim Phillips <kim.phillips@freescale.com>
Signed-off-by: Kumar Gala <galak@kernel.crashing.org>
Integrated-by: Andrew Liu <shengping.liu@windriver.com>
---
 drivers/crypto/caam/algapi.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/crypto/caam/algapi.c b/drivers/crypto/caam/algapi.c
index be9a44a..1491443 100644
--- a/drivers/crypto/caam/algapi.c
+++ b/drivers/crypto/caam/algapi.c
@@ -293,7 +293,7 @@ static int build_protocol_desc_ipsec_encap(struct caam_ctx *ctx,
 	 */
 
 #ifdef DEBUG
-	memcpy(&sh_desc.cbc.iv, "myivmyivmyivmyiv", sizeof(sh_desc.cbc.iv));
+	memcpy(&sh_desc->cbc.iv, "myivmyivmyivmyiv", sizeof(sh_desc->cbc.iv));
 #endif
 
 	/* indicate no IP header,
@@ -347,8 +347,8 @@ static int build_protocol_desc_ipsec_encap(struct caam_ctx *ctx,
 #ifdef DEBUG
 	print_hex_dump(KERN_ERR, "shrdesc@"xstr(__LINE__)": ",
 		       DUMP_PREFIX_ADDRESS, 16, 4, sh_desc,
-		       (sh_desc_ptr - sh_desc + 1 + 4) * 4, 1);
-	caam_desc_disasm(sh_desc, DISASM_SHOW_OFFSETS | DISASM_SHOW_RAW);
+			(sh_desc_ptr - (void *)sh_desc + 1 + 4) * 4, 1);
+	caam_desc_disasm((u32 *)sh_desc, DISASM_SHOW_OFFSETS | DISASM_SHOW_RAW);
 #endif
 
 	ctx->shared_desc_len = endidx * sizeof(u32);
-- 
1.6.5.2

